### ISSUE TITLE
Assertion violation at src/logics/Logic.cc:2040

### ISSUE THREAD

**Issue by rainoftime** [usi-verification-and-security/opensmt#124]
Hi, for this formula,
~~~~
(set-logic QF_UF)
(declare-fun _substvar_1828_ () Bool)
(declare-sort S0 0)
(declare-sort S1 0)
(declare-fun v0 () S0)
(declare-fun v1 () S1)
(declare-fun f0 (S1 S1) S0)
(declare-fun f1 (S0) S1)
(assert (not (distinct v0 (ite _substvar_1828_ (f0 (f1 v0) v1) v0))))
~~~~

opensmt 1a7b7241 throws an assertion failure
~~~~
/opensmt/src/logics/Logic.cc:2040: void Logic::conjoinItes(PTRef, PTRef&): Assertion `id < size' failed.
~~~~

**Comment by rainoftime** [comment]
Related issue: https://github.com/usi-verification-and-security/opensmt/issues/112

**Comment by blishko** [comment]
fixed by 9820755
